### PR TITLE
Made event watching a single process, added failover, and batched lists of events

### PR DESCRIPTION
**Pull request by raguiar2**

### CODE REVIEW

File: rockkube/watcher.py [hunk 1]
```diff
@@ -10,48 +10,72 @@
...
-    threads.append(thread)
-
-# We want the watchers to continue indefinitley, so we start them and run
-# them forever
-for thread in threads:
-    thread.start()
\ No newline at end of file
+    resource_version = batch_upload_events()
+    doc_buffer = []
+    for event in w.stream(v1.list_event_for_all_namespaces, _request_timeout=60, resource_version=resource_version):
```
Comment (dhruba) [inline rockkube/watcher.py]: can we make the default batching timeout by small (say 5 seconds) and then make it configurable/overridable via an environment variable?

**Comment by dhruba** [comment]
It will be great to put in code comments on how we use "resource_versions'.

**Comment by dhruba** [comment]
we need a good error messsage when we bail out aftre retrying 25 times. So that the user knows what error he/she is encountering.

File: rockkube/watcher.py (hunk 1 above)
Comment (raguiar2) [reply rockkube/watcher.py]: I can just add an argument with argparse and make it configurable from the cli

**Pull request merged**